File: code_snippets/api-guide/evaluate/evaluate_model.js
//index.js file

/////////////////////////////////////////////////////////////////////////////////////////////
// In this section, we set the user authentication, app ID, and model evaluation details.
// Change these strings to run your own example.
////////////////////////////////////////////////////////////////////////////////////////////

const USER_ID = "YOUR_USER_ID_HERE";
// Your PAT (Personal Access Token) can be found in the Account's Security section
const PAT = "YOUR_PAT_HERE";
const APP_ID = "YOUR_APP_ID_HERE";
// Change these to evaluate your own model
const MODEL_APP_ID = "YOUR_MODEL_APP_ID_HERE";
const MODEL_USER_ID = "YOUR_MODEL_USER_ID_HERE";
const MODEL_ID = "YOUR_MODEL_ID_HERE";
const MODEL_VERSION_ID = "YOUR_MODEL_VERSION_HERE";
const DATASET_USER_ID = "YOUR_DATASET_USER_ID_HERE";
const DATASET_APP_ID = "YOUR_DATASET_APP_ID_HERE";
const DATASET_ID = "YOUR_DATASET_ID_HERE";
const DATASET_VERSION_ID = "YOUR_DATASET_VERSION_ID_HERE";

///////////////////////////////////////////////////////////////////////////////////
// YOU DO NOT NEED TO CHANGE ANYTHING BELOW THIS LINE TO RUN THIS EXAMPLE
///////////////////////////////////////////////////////////////////////////////////

const { ClarifaiStub, grpc } = require("clarifai-nodejs-grpc");

const stub = ClarifaiStub.grpc();

// This will be used by every Clarifai endpoint call
const metadata = new grpc.Metadata();
metadata.set("authorization", "Key " + PAT);

stub.PostEvaluations(
  {
    user_app_id: {
      user_id: USER_ID,
      app_id: APP_ID,
    },
    eval_metrics: [
      {
        model: {
          app_id: MODEL_APP_ID,
          user_id: MODEL_USER_ID,
          id: MODEL_ID,
          model_version: {
            id: MODEL_VERSION_ID
          }
        },
        ground_truth_dataset: {
          user_id: DATASET_USER_ID,
          app_id: DATASET_APP_ID,
          id: DATASET_ID,
          version: {
            id: DATASET_VERSION_ID
          }
        }
      }
    ]

  },

  metadata,

  (err, response) => {
    if (err) {
      throw new Error(err);
    }

    if (response.status.code !== 10000) {
      console.error('Post models failed, status:', response.status);
      throw new Error("Evaluate model failed, status: " + response.status.description);
    }

  }
);

File: code_snippets/api-guide/evaluate/evaluate_model.java
package com.clarifai.example;

import com.clarifai.channel.ClarifaiChannel;
import com.clarifai.credentials.ClarifaiCallCredentials;
import com.clarifai.grpc.api.*;
import com.clarifai.grpc.api.status.StatusCode;

public class ClarifaiExample {

    //////////////////////////////////////////////////////////////////////////////////////////////
    // In this section, we set the user authentication, app ID, and model evaluation details.
    // Change these strings to run your own example.
    //////////////////////////////////////////////////////////////////////////////////////////////

    static final String USER_ID = "YOUR_USER_ID_HERE";
    // Your PAT (Personal Access Token) can be found in the portal under Authentification
    static final String PAT = "YOUR_PAT_HERE";
    static final String APP_ID = "YOUR_APP_ID_HERE";
    // Change these to evaluate your own model
    static final String MODEL_APP_ID = "YOUR_MODEL_APP_ID_HERE";
    static final String MODEL_USER_ID = "YOUR_MODEL_USER_ID_HERE";
    static final String MODEL_ID = "YOUR_MODEL_ID_HERE";
    static final String MODEL_VERSION_ID = "YOUR_MODEL_VERSION_HERE";
    static final String DATASET_USER_ID = "YOUR_DATASET_USER_ID_HERE";
    static final String DATASET_APP_ID = "YOUR_DATASET_APP_ID_HERE";
    static final String DATASET_ID = "YOUR_DATASET_ID_HERE";
    static final String DATASET_VERSION_ID = "YOUR_DATASET_VERSION_ID_HERE";

    ///////////////////////////////////////////////////////////////////////////////////
    // YOU DO NOT NEED TO CHANGE ANYTHING BELOW THIS LINE TO RUN THIS EXAMPLE
    ///////////////////////////////////////////////////////////////////////////////////
    
    public static void main(String[] args) {

        V2Grpc.V2BlockingStub stub = V2Grpc.newBlockingStub(ClarifaiChannel.INSTANCE.getGrpcChannel())
                .withCallCredentials(new ClarifaiCallCredentials(PAT));

        MultiEvalMetricsResponse postEvaluationsResponse = stub.postEvaluations(
                PostEvaluationsRequest.newBuilder()
                        .setUserAppId(UserAppIDSet.newBuilder().setUserId(USER_ID).setAppId(APP_ID))
                        .addEvalMetrics(
                                EvalMetrics.newBuilder()
                                        .setModel(Model.newBuilder()
                                                .setAppId(MODEL_APP_ID)
                                                .setUserId(MODEL_USER_ID)
                                                .setId(MODEL_ID)
                                                .setModelVersion(ModelVersion.newBuilder()
                                                        .setId(MODEL_VERSION_ID)
                                                )
                                        )
                                        .setGroundTruthDataset(Dataset.newBuilder()
                                                .setUserId(DATASET_USER_ID)
                                                .setAppId(DATASET_APP_ID)
                                                .setId(DATASET_ID)
                                                .setVersion(DatasetVersion.newBuilder()
                                                        .setId(DATASET_VERSION_ID)
                                                )
                                        )
                        )
                        .build()
        );

        if (postEvaluationsResponse.getStatus().getCode() != StatusCode.SUCCESS) {
            throw new RuntimeException("Evaluate model failed, status: " + postEvaluationsResponse.getStatus());
        }

    }

}
